setup.py

```diff
@@ -112,49 +134,75 @@ class BuildExtFailed(Exception):
     pass
 
 
-# Attempt to build a C-extension, catch and throw a common/custom error if there are any issues
+# Attempt to build a C-extension, catch exceptions so failed building skips the extension
+# DEV: This is basically what `distutils`'s' `Extension(optional=True)` does
 class optional_build_ext(build_ext):
     def run(self):
         try:
             build_ext.run(self)
-        except DistutilsPlatformError:
-            raise BuildExtFailed()
+        except DistutilsPlatformError as e:
+            extensions = [ext.name for ext in self.extensions]
+            print("WARNING: Failed to build extensions %r, skipping: %s" % (extensions, e))
 
     def build_extension(self, ext):
         try:
             build_ext.build_extension(self, ext)
-        except build_ext_errors:
-            raise BuildExtFailed()
+        except build_ext_errors as e:
+            print("WARNING: Failed to build extension %s, skipping: %s" % (ext.name, e))
 
 
-macros = []
-if sys.byteorder == "big":
-    macros = [("__BIG_ENDIAN__", "1")]
-else:
-    macros = [("__LITTLE_ENDIAN__", "1")]
+def get_msgpack_extensions():
+    try:
+        msgpack_setup = load_module_from_project_file("ddtrace.vendor.msgpack.setup", "ddtrace/vendor/msgpack/setup.py")
+        return msgpack_setup.get_extensions()
+    except Exception as e:
+        print("WARNING: Failed to load msgpack extensions, skipping: %s" % e)
+        return []
+
+
+def get_wrapt_extensions():
+    try:
+        wrapt_setup = load_module_from_project_file("ddtrace.vendor.wrapt.setup", "ddtrace/vendor/wrapt/setup.py")
+        return wrapt_setup.get_extensions()
+    except Exception as e:
+        print("WARNING: Failed to load wrapt extensions, skipping: %s" % e)
+        return []
+
+
+def get_psutil_extensions():
+    try:
+        psutil_setup = load_module_from_project_file("ddtrace.vendor.psutil.setup", "ddtrace/vendor/psutil/setup.py")
+        return psutil_setup.get_extensions()
+    except Exception as e:
+        print("WARNING: Failed to load psutil extensions, skipping: %s" % e)
+        return []
 
 
 # Try to build with C extensions first, fallback to only pure-Python if building fails
 try:
+    exts = []
+    msgpack_extensions = get_msgpack_extensions()
+    if msgpack_extensions:
+        exts.extend(msgpack_extensions)
+
+    wrapt_extensions = get_wrapt_extensions()
+    if wrapt_extensions:
+        exts.extend(wrapt_extensions)
+
+    psutil_extensions = get_psutil_extensions()
+    if psutil_extensions:
+        exts.extend(psutil_extensions)
+
```

Comment:
You could factor all of this in the end with a

```python
for extname in ("msgpack", "psutil", "wrapt"):
    ext = get_ext_for(extname)
    if ext:
        ext.extends(ext)
```

and write a `get_module_extensions(extname)`, no?

Reply:
yeah... that does seem better